servers/pollux/default.nix

```diff
@@ -0,0 +1,163 @@
+            mkdir -p "$_www_dir"
+            cp -Raf "${suitecrm}/." "$_www_dir/"
+
+            mkdir -p $_base_dir
+            for dir in ''${_persist_folders[@]}
+            do
+                mkdir -p "$_base_dir/$dir"
+                cp -Raf "${suitecrm}/$dir/." "$_base_dir/$dir/"
+                rm -r "$_www_dir/$dir"
+                ln -s "$_base_dir/$dir" "$_www_dir/$dir"
+            done
+
+            touch "$_base_dir/config.php"
+            ln -s "$_base_dir/config.php" "$_www_dir/config.php"
+          fi
+
+          chmod +rx /var/www
+          find /var/www/suitecrm -type f -exec chmod 640 {} \;
+          find /var/www/suitecrm -type d -exec chmod 750 {} \;
+          chown suitecrm:nginx -R "$_www_dir"
+          chown suitecrm:nginx -R "$_base_dir"
+        '';
+
+      serviceConfig.Type = "oneshot";
+    };
+
+  services.mysql = {
+    enable = true;
+
+    ensureUsers =
+      [ { name = "suitecrm";
+          ensurePermissions = {
+            "suitecrm.*" = "ALL PRIVILEGES";
+          };
+        }
+      ];
+
+    ensureDatabases =
+      [ "suitecrm"
+      ];
+  };
+
+  services.nginx = {
+    enable = true;
+    recommendedProxySettings = true;
+    virtualHosts.hermetic = {
+      default = true;
+
+      serverName = with config.networking; "${hostName}.${domain}";
+      enableACME = true;
+      forceSSL = true;
+
+      locations."/" = {
+        root = "/var/www/suitecrm";
+      };
+
+      locations."~ [^/]\.php(/|$)" = {
+        root = "/var/www/suitecrm";
+        extraConfig = ''
+          fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+          if (!-f $document_root$fastcgi_script_name) {
+              return 404;
+          }
+
+          include ${pkgs.nginx}/conf/fastcgi_params;
+          fastcgi_index index.php;
+          fastcgi_param  SCRIPT_FILENAME  $document_root$fastcgi_script_name;
+          fastcgi_pass unix:${config.services.phpfpm.pools.suitecrm.socket};
+        '';
+      };
+
+      # locations."/cache/" = {
+      #   root = "/var/suitecrm/cache/";
+      # };
+    };
+  };
+
+  services.phpfpm = {
+    pools.suitecrm = {
+      user = "suitecrm";
+      group = "nginx";
+
+      phpPackage = pkgs.php74.withExtensions ({ all, ... }: with all;
+        [ curl
+          intl
+          json
+          gd
+          mbstring
+          mysqli
+          pdo_mysql
+          openssl
+          soap
+          xml
+          zip
+          zlib
+          imap
+          session
+          ldap
+          pdo
+        ]);
+
+      settings = {
+        "pm" = "dynamic";
+        "pm.max_children" = 75;
+        "pm.start_servers" = 10;
+        "pm.min_spare_servers" = 5;
+        "pm.max_spare_servers" = 20;
+        "pm.max_requests" = 500;
+
+        "listen.owner" = "nginx";
+
+      };
+      phpOptions = ''
+        upload_max_filesize = 20M
+        post_max_size = 21M
+      '';
+    };
+  };
+
+  networking.hostName = "pollux";
+  wireguard-ip-address = "172.21.0.10";
```

Comment:
that doesn't seem right

Reply:
I haven't done any any of the wireguard and terraform parts, I can't test it, since I don't have access and also I've never worked with Terraform or AWS, should I attempt it nonetheless?

Reply:
Yep, never a better time to practice, I can pick it up instead if you have any issues. You should be able to use Terraform to deploy it to AWS, though updating our Wireguard config in Gemini will require us repinning so your key is added, I'll make sure that happens, but also I can deploy it instead.

Reply:
I'll try to conjure up something then

Reply:
just set the wireguard address to `172.21.0.33` as per what is available on `https://github.com/serokell/sagittarius-infra/blob/master/servers/polis/wireguard.nix` then we can open a PR to add it there once it is deployed and it generates itself a key

Reply:
done